client/blocks/payment-request/hooks.js

```diff
@@ -14,6 +14,8 @@ import {
 } from './event-handlers';
 import { createPaymentRequestUsingCart } from '../stripe-utils';
 
+const noop = { removeEventListener: ( event ) => void event };
+
```

Comment:
Out of curiosity; is there a reason you prefer to use a combination of `let`, `if`, and `noop` instead of the optional chaining?

I ask because I switched to the optional chaining because (IMO) the code was easier to understand when using optional chaining, and seeing you add it back into the code makes me wonder if there was a reason for using the "`noop`" approach in the first place 🤔 

If there are no architectural — or otherwise engineering related — differences between the two approaches, I think I'd prefer to keep the optional chaining for readability.

Reply:
I am not opinionated about this in any way, I was just basing my approach on the implementation in Blocks since the event handler implementation in this extension is broken. Personally, I see no difference in readability. If anything, the `noop` tells you what to expect in the object so really is no different. 

@nerrad may have more thoughts on this.

Reply:
"Readability" can sometimes be very subjective. There are times when I find optional chaining to be harder to follow :). There's definitely no architectural differences between the two approaches - mostly a preference thing.